Update debugging.md (#1323)
api-platform/docs: distribution/debugging.md

      # See https://docs.docker.com/docker-for-mac/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host
      # See https://github.com/docker/for-linux/issues/264
      # The `remote_host` below may optionally be replaced with `remote_connect_back`
      # XDEBUG_MODE required for step debugging
      XDEBUG_MODE: debug
      # default port for Xdebug 3 is 9003
      # idekey=VSCODE if you are debugging with VSCode
      XDEBUG_CONFIG: >-
        client_host=host.docker.internal

>         client_port=9003 

Reviewer: I don't think this line is necessary. This is the default port for XDebug 3.

Author: I'm happy to remove it, but the line I'm replacing `remote_port=9000` was also an unnecessary line, as the default port for Xdebug 2 was 9000 - I'm not sure if the intend was to just be informative?

Reviewer: I don't know but I think it's safe to remove it.

        remote_enable=1
        remote_host=docker.for.mac.localhost
        remote_connect_back=0
        remote_port=9003
        idekey=PHPSTORM
        remote_autostart=1
        remote_mode=req
        remote_handler=dbgp
```

> In VSCode, alongside the default php config in `launch.json`, you'll need path mappings for a docker image.

Reviewer: I don't think it's a good idea to add IDE specific related configuration.

Author: I think for 99% of the docs that is definitely the case, but in the case of debugging, particularly step debugging, it is inherently tied to an IDE. Before my modifications there were already specifics things to php storm in this doc, and there are mentions of php storm and vs code in the recommended IDEs for API Platform. Docker is also recommended, and this path mapping is a quirk of docker and API platform together. 

If the intent is to be IDE agnostic, then there likely shouldn't be recommendations to use php storm and vs code elsewhere.

For me, knowing this config info would have saved me a few hours, and this doesn't clutter up the docs much.

Reviewer 2: I'm in favor of having docs for VSCode and PHPStorm as it's the two most popular tools in the PHP world.

Reviewer: Ok then :smile:

> In VSCode, alongside the default php config in `launch.json`, you'll need path mappings for a docker image.

Reviewer: ```suggestion
In VSCode, alongside the default PHP configuration in `launch.json`, you'll need path mappings for the Docker image.
```

```json
{
    "version": "0.2.0",
    "configurations": [
        {

>             "name": "Listen for Xdebug",

Reviewer: ```suggestion
            "name": "Listen for XDebug",
```

            "type": "php",
            "request": "launch",
            "port": 9003,
            "log": true,
            "pathMappings": {
                "/srv/api": "${workspaceFolder}/api"
            },
        },
    ]
}
```

> Note for Linux environments, the `client_host` setting of `host.docker.internal` will not work, you will need the actual local IP address of your computer.

Reviewer: ```suggestion

Note: For Linux environments, the `client_host` setting of `host.docker.internal` will not work, you will need the actual local IP address of your computer.
```

    with Xdebug v3.0.2, Copyright (c) 2002-2021, by Derick Rethans
    …
```

> 

Reviewer 2: These extra blank lines must be removed.
